server/src/main/java/org/elasticsearch/search/fetch/PreloadedFieldLookupProvider.java

```diff
@@ -32,12 +33,18 @@ class PreloadedFieldLookupProvider implements LeafFieldLookupProvider {
 
     @Override
     public void populateFieldLookup(FieldLookup fieldLookup, int doc) throws IOException {
-        String field = fieldLookup.fieldType().name();
+        if (storedFields == null) {
+            storedFields = Collections.emptyMap();
```

Comment:
Is this the actual fix in this change?

Reply:
Yes...the thing is that `storedFields` there is a cache if I understand correctly. SO having it null actually means it is empty because it has not been loaded...not sure if the expectation is that it is always loaded before the method is used.

Reply:
I think loading fields directly from the Lucene segment is correct instead of using an empty map which would result in a cache coherency issue because it would not match fields available in the Lucene segment.